File: backend/schemas/tools.py
from pydantic import BaseModel, Field
from typing import Optional, Any, List


class ExportArticlesRequest(BaseModel):
    """导出文章请求模型"""

    mp_id: str = Field(..., description="公众号ID")
    doc_id: Optional[List[str]] = Field(None, description="文档ID列表, 为空则导出所有文章")
    page_size: int = Field(10, description="每页数量", ge=1, le=10)
    page_count: int = Field(1, description="页数, 0表示全部", ge=0, le=10000)
    add_title: bool = Field(True, description="是否添加标题")
    remove_images: bool = Field(True, description="是否移除图片")
    remove_links: bool = Field(False, description="是否移除链接")
    export_md: bool = Field(False, description="是否导出Markdown格式")
    export_docx: bool = Field(False, description="是否导出Word文档格式")
    export_json: bool = Field(False, description="是否导出JSON格式")
    export_csv: bool = Field(False, description="是否导出CSV格式")
    export_pdf: bool = Field(True, description="是否导出PDF格式")
    zip_filename: Optional[str] = Field(None, description="压缩包文件名，为空则自动生成")


class ExportArticlesResponse(BaseModel):
    """导出文章响应模型"""

    record_count: int = Field(..., description="导出的文章数量")
    export_path: str = Field(..., description="导出文件路径")
    message: str = Field(..., description="导出结果消息")


class ExportFileInfo(BaseModel):
    """导出文件信息模型"""

    filename: str = Field(..., description="文件名")
    size: int = Field(..., description="文件大小（字节）")
    created_time: str = Field(..., description="创建时间（ISO格式）")
    modified_time: str = Field(..., description="修改时间（ISO格式）")


class DeleteFileRequest(BaseModel):
    """删除文件请求模型"""

    filename: str = Field(..., description="文件名")
    mp_id: str = Field(..., description="公众号ID")

File: frontend/src/api/tools.ts
import http from './http'

export interface ExportArticlesParams {
  mp_id: string
  scope?: 'selected' | 'all'
  ids?: string[]
  limit?: number
  page_count?: number
  add_title?: boolean
  remove_images?: boolean
  remove_links?: boolean
  format: string[]
  zip_filename?: string
}

export interface ExportRecordsParams {
  mp_id: string
}

export interface DeleteExportRecordsParams {
  mp_id?: string
  filename: string
}

export const exportArticles = (params: ExportArticlesParams) => {
  const requestData = {
    mp_id: params.mp_id,
    doc_id: params.scope === 'selected' ? params.ids ?? [] : [],
    page_size: params.limit ?? 10,
    page_count: params.page_count ?? 1,
    // 默认为添加标题；显式传 false 时要保留 false
    add_title: params.add_title ?? true,
    remove_images: params.remove_images ?? false,
    remove_links: params.remove_links ?? false,
    export_md: params.format?.includes('md') ?? false,
    export_docx: params.format?.includes('docx') ?? false,
    export_json: params.format?.includes('json') ?? false,
    export_csv: params.format?.includes('csv') ?? false,
    export_pdf: params.format?.includes('pdf') ?? false,
    zip_filename: params.zip_filename ?? '',
  }
  return http.post<{ code: number; data: string }>('/wx/tools/export/articles', requestData, {
    headers: {
      'Content-Type': 'application/json',
      Accept: 'application/json',
      'X-Requested-With': 'XMLHttpRequest',
    },
  })
}

export const getExportRecords = (params: ExportRecordsParams) => {
  const requestData = {
    mp_id: params.mp_id,
  }
  return http.get<{ code: number; data: string }>('/wx/tools/export/list', {
    params: requestData,
  })
}

export const DeleteExportRecords = (params: DeleteExportRecordsParams) => {
  const requestData = {
    mp_id: params.mp_id ?? '',
    filename: params.filename,
  }
  return http.delete<{ code: number; data: string }>('/wx/tools/export/delete', {
    data: requestData,
  })
}
